packages/core/src/Site/index.js

```diff
@@ -816,6 +819,59 @@ class Site {
     }
   }
 
+  async reloadSiteConfig() {
+    const oldAddressablePages = this.addressablePages.slice();
+    const oldPagesSrc = oldAddressablePages.map(page => page.src);
+    await this.readSiteConfig();
+    this.updateAddressablePages();
+    // Get pages with edited attributes but with the same src
+    const editedPages = _.differenceWith(this.addressablePages, oldAddressablePages, (newPage, oldPage) => {
+      if (!_.isEqual(newPage, oldPage)) {
+        return !oldPagesSrc.includes(newPage.src);
+      }
+      return true;
+    });
+
+    // Comparator for the _differenceWith comparison below
+    const isNewPage = (newPage, oldPage) => _.isEqual(newPage, oldPage) || newPage.src === oldPage.src;
+
+    const addedPages = _.differenceWith(this.addressablePages, oldAddressablePages, isNewPage);
+    const removedPages = _.differenceWith(oldAddressablePages, this.addressablePages, isNewPage)
+      .map(filePath => Site.setExtension(filePath.src, '.html'));
+    if (!_.isEmpty(addedPages) || !_.isEmpty(removedPages)) {
+      await this.removeAsset(removedPages);
+      await this.rebuildSourceFiles();
+      await this.writeSiteData();
+    } else {
+      this.updatePages(editedPages);
+      const siteConfigDirectory = path.dirname(path.join(this.rootPath, this.siteConfigPath));
+      this.rebuildAffectedSourceFiles(editedPages.map(page => path.join(siteConfigDirectory, page.src)));
+    }
+  }
+
+  /**
+   * Creates new pages and replaces the original pages with the updated version
+   */
+  updatePages(pagesToUpdate) {
+    pagesToUpdate.forEach((pageToUpdate) => {
+      this.pages.forEach((page, index) => {
+        if (page.pageConfig.src === pageToUpdate.src) {
+          const newPage = this.createPage({
```

Comment:
how about something like a `mapAddressablePageToPage` function to reduce duplication with `mapAddressablePagesToPages` (or if could generalise the latter even better)

Reply:
Hmm I tried to combine them but it seems like their logic is quite different. As a workaround, I've extracted out the `createPage` part into a separate function so that both `mapAddressablePagesToPages` and this can call the new function which resolves the repetition issue. Is this ok?